src/Core/Content/Product/Cart/ProductCartProcessor.php

```diff
@@ -326,21 +326,24 @@ private function enrich(SalesChannelContext $context, LineItem $lineItem, CartDa
             );
         }
 
-        $quantityInformation = new QuantityInformation();
+        //prevent quantity informations from being overwritten
```

Comment:
hey @ufcyg,

We can't merge this changes because it is in conflict with the following code:
https://github.com/shopware/platform/blob/trunk/src/Core/Content/Product/Subscriber/ProductSubscriber.php#L145

There we calculate the max allowed purchase and consider the available stock. Means if the product goes out of stock or the stock is reduced, the max purchase will be updated. 

Now you only update the quantity information one time and if the items remains for a while this information will never be updated.

If you want to overwrite the quantity information for a nested line item, you have to handle this inside an own processor. Quantity information and price information are critical information which has to be validated from our side